### ISSUE TITLE
Persistent Search Bar

### ISSUE THREAD

**Issue by SiddhantRakh** [Malopieds/InnerTune#26]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

Search something. Scroll to view more results. The search bar stays on top. 

### Expected behavior

The search bar should hide as it does on the home screen. 

### Actual behavior

The search bar persists looking a bit odd. 

### Screenshots/Screen recordings

![Screenshot_20240329-200117_InnerTune](https://github.com/Malopieds/InnerTune/assets/42494931/26cdb37c-1700-456c-9260-d6be06b021e0)
![Screenshot_20240329-200127_InnerTune](https://github.com/Malopieds/InnerTune/assets/42494931/ca06704e-1102-4683-aaa6-ac921461621c)

### Logs

Sorry I can't figure out how to provide logs. 

### InnerTune version

0.5.5

### Android version

Android 12

**Comment by Malopieds** [comment]
Good catch, I'll try to investigate !

**Comment by W1zardK1ng** [comment]
Just a suggestion. If we could make it search bar section persistent with dark background, we can access the options below the search(All, songs, videos etc) even if we scrolled down.

**Comment by Malopieds** [comment]
Well, there are 2 options, we either:
- keep the idea where the search bar and filter are hidden when scrolling down, but if we scroll up both reappears (like in the home screen for example)
- we make it always accessible but with a background not transparent

**Comment by SiddhantRakh** [comment]
> Well, there are 2 options, we either:
> 
> * keep the idea where the search bar and filter are hidden when scrolling down, but if we scroll up both reappears (like in the home screen for example)
> * we make it always accessible but with a background not transparent

Well I personally feel that making it always accessible is better option as most of the options of search are associated with the search bar and having it stay at top with a nice fade behind will definitely be better both functionally and visually. 

BUT I strongly recommend implementing this behaviour everywhere inside the app i.e. on both home screen and the search screen is the way to go. As most of the sorting options in the artists and playlists tab makes more sense to be visible all the time. 

Lastly whatever option to choose is your preference but whatever gets implemented should be done on both home and search screen just for the same of simplicity and uniformity.

**Comment by Malopieds** [comment]
Yes I agree with you on the fact that it should be consistent. I will try to implement it as soon as I can.

**Comment by SiddhantRakh** [comment]
Thanks. I hope that the suggestions I make turn out to be helpful for the app.
